__tests__/nc_news_test.test.js

```diff
@@ -356,4 +356,29 @@ describe('GET /api/topics', () => {
                 expect(response.body.msg).toBe("Bad Request")
             })
         })
-    });
\ No newline at end of file
+    });
+    describe('GET /api/users', () => {
+        test('GET /api/users responds with an array of objects, each possessing username, name and avatar_url properties', () => {
+            return request(app)
+            .get("/api/users")
+            .expect(200)
+            .then((response)=> {
+                const users = response.body.users
+                
+                expect(users.length).toBe(4)
+                users.forEach((user)=>{
+                    expect(user).toHaveProperty("username");
+                    expect(user).toHaveProperty("name");
+                    expect(user).toHaveProperty("avatar_url");
+                })
+            })
```

Comment:
This test is _good_, it's checking that the array has the expected length of 4, and then also checking that the objects have the expected properties 👍 

To make this test _great_ you could adjust your assertions to check the property values as well.  Instead of `toHaveProperty` you could do something like:
`expect(typeof user.name).toBe("string");` for each property

Why is this more robust? Because it's checking two things:
- it's checking that the property _exists_, otherwise it couldn't be a string
- and it's checking that the value is some kind of _string_, otherwise for all we know every value could just be `null`